app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,20 @@
+class CustomersController < ApplicationController
+  def index
+
+  end
+  
+  def show
+    @customer = Customer.find(params[:id])
+
+    if request.post? && params[:customer].present? && params[:customer][:item_id].present?
```

Comment:
While it is okay for one controller action to receipt multiple types of requests, this usually only happens when those requests are really similar. In this case, I wouldn't have your form redirect back to this show action. It's not the most RESTful option because the `show` action is generally just a data displaying action (i.e. grab some data from the db and hand it to the view) but what you're doing here is `creating` something -- a relationship between a customer and an item. 

So instead of a `show` action, we want a `create` action!

Where to put the `create` action is a different question. If we put it in this same CustomersController, it suggests that we are creating a new customer. But we're not creating a new customer but just a new relationship between existing customers and items. So, what is conventional here is to create a new *joins* controller and call it CustomerItemsController (or ItemCustomersController.. doesn't matter). If you route that form to "customer_items#create", that is the most RESTful option to indicate that you're creating a joins record. 

This would also remove the need to check if the request is a post, but nice work setting up this conditional logic!! At the end of the `create` action in the CustomerItemsController, you can redirect the request to "/customers/:id" to get back to the show page